lib/ApplicationFeatures/ApplicationServer.cpp

```diff
@@ -68,16 +69,27 @@ static void failCallback(std::string const& message) {
       << "error. cannot proceed. reason: " << message;
   FATAL_ERROR_EXIT();
 }
+
+auto nonEmptyFeatures(
+    std::span<std::unique_ptr<ApplicationFeature>> features) noexcept {
+  return boost::make_iterator_range(features) |
```

Comment:
Happy when this can be changed to ranges eventually, but they don't seem to be supported well by compilers right now.